### ISSUE TITLE
Error: "Missing PDF file"

### ISSUE THREAD

**Issue by Avrantinis** [nextcloud/files_pdfviewer#391]
### Steps to reproduce
1. Click on a pdf file
2.
3.

### Expected behaviour

Open pdf file in a frame in the same window

### Actual behaviour

Opened an empty frame "Missing PDF file"
"More Information"
PDF.js v2.5.207 (build: 0974d6052)
Message: Missing PDF "https://cloud.athensmusic.gr/remote.php/dav/files/admin/pdf_file_name.pdf

### Server configuration

**Operating system**:

**Web server:**

**Database:**

**PHP version:**

**Nextcloud version:** (see Nextcloud admin page)
Worked as expected on previous versions 19, 20, 21.0.1 etc untill latest update
21.0.2

**Where did you install Nextcloud from:**
Cpanel Softaculous

**List of activated apps:**
Enabled:
  - activity: 2.14.3
  - admin_audit: 1.11.0
  - apporder: 0.12.0
  - audioplayer: 3.1.0
  - cloud_federation_api: 1.4.0
  - comments: 1.11.0
  - dav: 1.17.1
  - deck: 1.4.2
  - duplicatefinder: 0.0.8
  - external: 3.8.1
  - extract: 1.3.1
  - federatedfilesharing: 1.11.0
  - federation: 1.11.0
  - files: 1.16.0
  - files_accesscontrol: 1.11.0
  - files_automatedtagging: 1.11.0
  - files_downloadactivity: 1.10.0
  - files_fulltextsearch: 21.0.2
  - files_linkeditor: 1.1.5
  - files_markdown: 2.3.3
  - files_pdfviewer: 2.1.0
  - files_retention: 1.10.1
  - files_rightclick: 1.0.0
  - files_sharing: 1.13.1
  - files_texteditor: 2.14.0
  - files_trackdownloads: 1.10.0
  - files_trashbin: 1.11.0
  - files_versions: 1.14.0
  - files_videoplayer: 1.10.0
  - fulltextsearch: 21.0.1
  - logreader: 2.6.0
  - lookup_server_connector: 1.9.0
  - nextcloud_announcements: 1.10.0
  - notifications: 2.9.0
  - oauth2: 1.9.0
  - password_policy: 1.11.0
  - photos: 1.3.0
  - privacy: 1.5.0
  - provisioning_api: 1.11.0
  - recommendations: 1.0.0
  - serverinfo: 1.11.0
  - settings: 1.3.0
  - sharebymail: 1.11.0
  - sharingpath: 0.3.0
  - support: 1.4.0
  - survey_client: 1.9.0
  - systemtags: 1.11.0
  - text: 3.2.0
  - theming: 1.12.0
  - theming_customcss: 1.8.0
  - twofactor_backupcodes: 1.10.0
  - updatenotification: 1.11.0
  - viewer: 1.5.0
  - workflow_script: 1.6.0
  - workflowengine: 2.3.0
Disabled:
  - accessibility
  - contactsinteraction
  - dashboard
  - encryption
  - files_external
  - firstrunwizard
  - user_ldap
  - user_status
  - weather_status

**Nextcloud configuration:**
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "cloud.athensmusic.gr"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "21.0.2.1",
        "overwrite.cli.url": "https:\/\/cloud.athensmusic.gr",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "mail_smtpmode": "smtp",
        "mail_smtpsecure": "ssl",
        "mail_sendmailmode": "smtp",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "maintenance": false,
        "filelocking.enabled": true,
        "updater.release.channel": "stable",
        "mysql.utf8mb4": true,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "default_phone_region": "+30",
        "theme": "",
        "loglevel": 2,
        "app_install_overwrite": [
            "files_linkeditor"
        ],
        "updater.secret": "***REMOVED SENSITIVE VALUE***"
    }
}

### Client configuration
**Browser:**
Chrome

**Operating system:**
Windows 10

#### Nextcloud log (data/owncloud.log)
Log is too large so I copied last two rows 
```
{"reqId":"YKizb9rG3ZVCiWUYfqK9NgAAAQg","level":3,"time":"2021-05-22T07:31:59+00:00","remoteAddr":"37.6.3.1","user":"admin","app":"PHP","method":"PROPFIND","url":"/remote.php/dav/files/admin/MASTER%20OFFICE","message":{"Exception":"Error","Message":"include(): Failed opening '/home/at383439/cloud.athensmusic.gr/3rdparty/composer/../icewind/searchdav/src/Backend/ISearchBackend.php' for inclusion (include_path='/home/at383439/cloud.athensmusic.gr/3rdparty/pear/archive_tar:/home/at383439/cloud.athensmusic.gr/3rdparty/pear/console_getopt:/home/at383439/cloud.athensmusic.gr/3rdparty/pear/pear-core-minimal/src:/home/at383439/cloud.athensmusic.gr/3rdparty/pear/pear_exception:/home/at383439/cloud.athensmusic.gr/apps') at /home/at383439/cloud.athensmusic.gr/lib/composer/composer/ClassLoader.php#478","Code":0,"Trace":[{"file":"/home/at383439/cloud.athensmusic.gr/lib/composer/composer/ClassLoader.php","line":478,"function":"onError","class":"OC\\Log\\ErrorHandler","type":"::","args":[2,"include(): Failed opening '/home/at383439/cloud.athensmusic.gr/3rdparty/composer/../icewind/searchdav/src/Backend/ISearchBackend.php' for inclusion (include_path='/home/at383439/cloud.athensmusic.gr/3rdparty/pear/archive_tar:/home/at383439/cloud.athensmusic.gr/3rdparty/pear/console_getopt:/home/at383439/cloud.athensmusic.gr/3rdparty/pear/pear-core-minimal/src:/home/at383439/cloud.athensmusic.gr/3rdparty/pear/pear_exception:/home/at383439/cloud.athensmusic.gr/apps')","/home/at383439/cloud.athensmusic.gr/lib/composer/composer/ClassLoader.php",478,{"file":"/home/at383439/cloud.athensmusic.gr/3rdparty/composer/../icewind/searchdav/src/Backend/ISearchBackend.php"}]},{"file":"/home/at383439/cloud.athensmusic.gr/lib/composer/composer/ClassLoader.php","line":478,"function":"include"},{"file":"/home/at383439/cloud.athensmusic.gr/lib/composer/composer/ClassLoader.php","line":346,"function":"Composer\\Autoload\\includeFile","args":["/home/at383439/cloud.athensmusic.gr/3rdparty/composer/../icewind/searchdav/src/Backend/ISearchBackend.php"]},{"function":"loadClass","class":"Composer\\Autoload\\ClassLoader","type":"->","args":["SearchDAV\\Backend\\ISearchBackend"]},{"file":"/home/at383439/cloud.athensmusic.gr/apps/dav/lib/Files/LazySearchBackend.php","line":29,"function":"spl_autoload_call","args":["SearchDAV\\Backend\\ISearchBackend"]},{"file":"/home/at383439/cloud.athensmusic.gr/lib/composer/composer/ClassLoader.php","line":478,"args":["/home/at383439/cloud.athensmusic.gr/apps/dav/lib/Files/LazySearchBackend.php"],"function":"include"},{"file":"/home/at383439/cloud.athensmusic.gr/lib/composer/composer/ClassLoader.php","line":346,"function":"Composer\\Autoload\\includeFile","args":["/home/at383439/cloud.athensmusic.gr/apps/dav/composer/composer/../../lib/Files/LazySearchBackend.php"]},{"function":"loadClass","class":"Composer\\Autoload\\ClassLoader","type":"->","args":["OCA\\DAV\\Files\\LazySearchBackend"]},{"file":"/home/at383439/cloud.athensmusic.gr/apps/dav/lib/Server.php","line":225,"function":"spl_autoload_call","args":["OCA\\DAV\\Files\\LazySearchBackend"]},{"file":"/home/at383439/cloud.athensmusic.gr/apps/dav/appinfo/v2/remote.php","line":34,"function":"__construct","class":"OCA\\DAV\\Server","type":"->","args":[{"__class__":"OC\\AppFramework\\Http\\Request"},"/remote.php/dav/"]},{"file":"/home/at383439/cloud.athensmusic.gr/remote.php","line":167,"args":["/home/at383439/cloud.athensmusic.gr/apps/dav/appinfo/v2/remote.php"],"function":"require_once"}],"File":"/home/at383439/cloud.athensmusic.gr/lib/private/Log/ErrorHandler.php","Line":92,"CustomMessage":"--"},"userAgent":"Mozilla/5.0 (Windows) mirall/3.2.1stable-Win64 (build 20210428) (Nextcloud, windows-10.0.19042 ClientArchitecture: x86_64 OsArchitecture: x86_64)","version":"21.0.2.1"}
{"reqId":"YKizb9rG3ZVCiWUYfqK9NgAAAQg","level":4,"time":"2021-05-22T07:31:59+00:00","remoteAddr":"37.6.3.1","user":"admin","app":"webdav","method":"PROPFIND","url":"/remote.php/dav/files/admin/MASTER%20OFFICE","message":{"Exception":"Sabre\\DAV\\Exception\\ServiceUnavailable","Message":"Error: Interface 'SearchDAV\\Backend\\ISearchBackend' not found","Code":0,"Trace":[{"file":"/home/at383439/cloud.athensmusic.gr/3rdparty/sabre/event/lib/WildcardEmitterTrait.php","line":89,"function":"{closure}","args":["*** sensitive parameters replaced ***"]},{"file":"/home/at383439/cloud.athensmusic.gr/3rdparty/sabre/dav/lib/DAV/Server.php","line":456,"function":"emit","class":"Sabre\\DAV\\Server","type":"->","args":["beforeMethod:PROPFIND",["*** sensitive parameter replaced ***","*** sensitive parameter replaced ***"]]},{"file":"/home/at383439/cloud.athensmusic.gr/3rdparty/sabre/dav/lib/DAV/Server.php","line":253,"function":"invokeMethod","class":"Sabre\\DAV\\Server","type":"->","args":["*** sensitive parameter replaced ***","*** sensitive parameter replaced ***"]},{"file":"/home/at383439/cloud.athensmusic.gr/3rdparty/sabre/dav/lib/DAV/Server.php","line":321,"function":"start","class":"Sabre\\DAV\\Server","type":"->","args":[]},{"file":"/home/at383439/cloud.athensmusic.gr/remote.php","line":76,"function":"exec","class":"Sabre\\DAV\\Server","type":"->","args":[]},{"file":"/home/at383439/cloud.athensmusic.gr/remote.php","line":171,"function":"handleException","args":[{"__class__":"Error"}]}],"File":"/home/at383439/cloud.athensmusic.gr/remote.php","Line":74,"CustomMessage":"--"},"userAgent":"Mozilla/5.0 (Windows) mirall/3.2.1stable-Win64 (build 20210428) (Nextcloud, windows-10.0.19042 ClientArchitecture: x86_64 OsArchitecture: x86_64)","version":"21.0.2.1"}
```

#### Browser log
```
Failed to load resource: the server responded with a status of 404 ()
excludeClickOutsideClasses.js:1 Uncaught TypeError: Cannot read property 'contains' of undefined
    at a.hasNodeOrAnyParentClass (excludeClickOutsideClasses.js:22)
    at a.clickOutsideMiddleware (excludeClickOutsideClasses.js:22)
    at r (v-click-outside.umd.js:1)
    at v-click-outside.umd.js:1
Coursera%25207RWYWAYBAJCN.pdf:1 Failed to load resource: the server responded with a status of 404 ()
cloud.athensmusic.gr/:1 Uncaught (in promise) MissingPDFException: Missing PDF "https://cloud.athensmusic.gr/remote.php/dav/files/admin/MASTER%2520OFFICE/Coursera%25207RWYWAYBAJCN.pdf".
excludeClickOutsideClasses.js:1 Uncaught TypeError: Cannot read property 'contains' of undefined
    at a.hasNodeOrAnyParentClass (excludeClickOutsideClasses.js:22)
    at a.clickOutsideMiddleware (excludeClickOutsideClasses.js:22)
    at r (v-click-outside.umd.js:1)
    at v-click-outside.umd.js:1
/remote.php/dav/files/admin/MASTER%2520OFFICE/Coursera%2520JWKGTLFC9V7G.pdf:1 Failed to load resource: the server responded with a status of 404 ()
/index.php/apps/file…0JWKGTLFC9V7G.pdf:1 Uncaught (in promise) MissingPDFException: Missing PDF "https://cloud.athensmusic.gr/remote.php/dav/files/admin/MASTER%2520OFFICE/Coursera%2520JWKGTLFC9V7G.pdf".
excludeClickOutsideClasses.js:22 Uncaught TypeError: Cannot read property 'contains' of undefined
    at a.hasNodeOrAnyParentClass (excludeClickOutsideClasses.js:22)
    at a.clickOutsideMiddleware (excludeClickOutsideClasses.js:22)
    at r (v-click-outside.umd.js:1)
    at v-click-outside.umd.js:1
hasNodeOrAnyParentClass @ excludeClickOutsideClasses.js:22
clickOutsideMiddleware @ excludeClickOutsideClasses.js:22
r @ v-click-outside.umd.js:1
(anonymous) @ v-click-outside.umd.js:1
setTimeout (async)
el @ v-click-outside.umd.js:1
handler @ v-click-outside.umd.js:1
(anonymous) @ PDFView.vue:76
(anonymous) @ vue.runtime.esm.js:1980
Kt @ vue.runtime.esm.js:1906
Promise.then (async)
Yt @ vue.runtime.esm.js:1933
ee @ vue.runtime.esm.js:1990
t.$nextTick @ vue.runtime.esm.js:3521
(anonymous) @ PDFView.vue:75
A @ runtime.js:63
(anonymous) @ runtime.js:293
(anonymous) @ runtime.js:118
o @ files_pdfviewer-main.js?v=f9249c0c-6:1
a @ files_pdfviewer-main.js?v=f9249c0c-6:1
(anonymous) @ files_pdfviewer-main.js?v=f9249c0c-6:1
(anonymous) @ files_pdfviewer-main.js?v=f9249c0c-6:1
mounted @ files_pdfviewer-main.js?v=f9249c0c-6:1
$t @ vue.runtime.esm.js:1854
Je @ vue.runtime.esm.js:4219
insert @ vue.runtime.esm.js:3139
_ @ vue.runtime.esm.js:6346
(anonymous) @ vue.runtime.esm.js:6565
t._update @ vue.runtime.esm.js:3948
r @ vue.runtime.esm.js:4066
pn.get @ vue.runtime.esm.js:4479
pn.run @ vue.runtime.esm.js:4554
cn @ vue.runtime.esm.js:4310
(anonymous) @ vue.runtime.esm.js:1980
Kt @ vue.runtime.esm.js:1906
Promise.then (async)
Yt @ vue.runtime.esm.js:1933
ee @ vue.runtime.esm.js:1990
(anonymous) @ vue.runtime.esm.js:4402
pn.update @ vue.runtime.esm.js:4544
ct.notify @ vue.runtime.esm.js:730
set @ vue.runtime.esm.js:1055
dn.set @ vue.runtime.esm.js:4631
(anonymous) @ Viewer.vue:430
A @ runtime.js:63
(anonymous) @ runtime.js:293
(anonymous) @ runtime.js:118
k @ viewer-main.js?v=f9249c0c-6:14
a @ viewer-main.js?v=f9249c0c-6:14
Promise.then (async)
k @ viewer-main.js?v=f9249c0c-6:14
a @ viewer-main.js?v=f9249c0c-6:14
(anonymous) @ viewer-main.js?v=f9249c0c-6:14
(anonymous) @ viewer-main.js?v=f9249c0c-6:14
(anonymous) @ Viewer.vue:349
file @ Viewer.vue:274
pn.run @ vue.runtime.esm.js:4568
cn @ vue.runtime.esm.js:4310
(anonymous) @ vue.runtime.esm.js:1980
Kt @ vue.runtime.esm.js:1906
Promise.then (async)
Yt @ vue.runtime.esm.js:1933
ee @ vue.runtime.esm.js:1990
(anonymous) @ vue.runtime.esm.js:4402
pn.update @ vue.runtime.esm.js:4544
ct.notify @ vue.runtime.esm.js:730
set @ vue.runtime.esm.js:1055
value @ Viewer.js:137
e.default @ FilesActionHandler.js:36
action @ merged-index.js?v=f9249c0c-6:2961
_onClickFile @ merged-index.js?v=f9249c0c-6:4832
Xe @ _executeBound.js:8
(anonymous) @ bind.js:10
(anonymous) @ restArguments.js:16
dispatch @ jquery.js:5183
g.handle @ jquery.js:4991
fetch_stream.js:128 GET https://cloud.athensmusic.gr/remote.php/dav/files/admin/MASTER%2520OFFICE/Coursera%25207RWYWAYBAJCN.pdf 404
PDFFetchStreamReader @ fetch_stream.js:128
getFullReader @ fetch_stream.js:78
(anonymous) @ api.js:2010
(anonymous) @ message_handler.js:371
_createStreamSink @ message_handler.js:370
MessageHandler._onComObjOnMessage @ message_handler.js:144
cloud.athensmusic.gr/:1 Uncaught (in promise) MissingPDFException {message: "Missing PDF "https://cloud.athensmusic.gr/remote.p…MASTER%2520OFFICE/Coursera%25207RWYWAYBAJCN.pdf".", name: "MissingPDFException"}

**Comment by beardhatcode** [comment]
Hi @Avrantinis thank you for reporting this, This is however a duplicate of #381, a "quickfix" can be found in that issue, and a real fix is on its way in PR #384 . 

Duplicate of #381

**Comment by beardhatcode** [comment]
Duplicate of #381
